cpp/src/strings/char_types/char_types.cu

```diff
@@ -79,6 +80,55 @@ std::unique_ptr<column> all_characters_of_type( strings_column_view const& strin
     return results;
 }
 
+std::unique_ptr<column> is_integer( strings_column_view const& strings,
```

Comment:
I wonder if it's valuable to provide `all_integer` and `all_float` versions of these that replace the `transform` with `transform_reduce` where the reduction functor is a Boolean AND? 

This would be more efficient than `is_integer` followed by an AND reduction, so if it's useful for Python when deciding whether it's safe to call `to_integer`, it might be worthwhile. 

Just a thought.

Reply:
So Pandas specifies the specific failing string:
```
>>> import pandas as pd
>>> ps = pd.Series(["1","hello","2","3"])
>>> ps.astype('int')
ValueError: invalid literal for int() with base 10: 'hello'
```
So I want to be able to give the Python cudf the ability to report a similar error by identifying which strings fail. 

It is easy enough to include both sets of APIs and let the caller choose.